File: LeetCode/Easy/1957. Delete Characters to Make Fancy String/2024-01-21_Accepted.py
class Solution:
  def makeFancyString(self, s: str) -> str:
    result = []
    pre = ''
    count = 0
    for c in s:
      if c != pre:
        pre = c
        count = 0
      count += 1
      if count <= 2:
        result.append(c)
    return ''.join(result)

File: LeetCode/Easy/1952. Three Divisors/2024-01-21_Accepted.cpp
class Solution {
public:
  bool isThree(int n) {
    int count = 0;
    for (int i = 1; i * i <= n; i++) {
      if (n % i == 0) {
        if (i * i == n) {
          count++;
        } else {
          count += 2;
        }
        if (count > 3) {
          return false;
        }
      }
    }
    return count == 3;
  }
};
